### PR TITLE
 Implement Mobile-Menu Features and Page Navigation

### PR DESCRIPTION
**Pull request by Peter-Kibet**
This pull request implements mobile menu features and enhances page navigation for mobile devices. It was a collaborative effort between myself and my partner, Zuheb. The following features have been implemented:

**1. Mobile Menu Display:**
- When the user clicks (or taps) the hamburger button, the mobile menu appears.
- When the user clicks (or taps) the close (X) button, the mobile menu disappears.
 
**2. Mobile Menu Navigation:**
- When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears.
- Additionally, the corresponding section of the page is displayed.

### **Changes Made:**
1. Added JavaScript event listeners to handle mobile menu interactions.
2. Updated the CSS to ensure proper positioning and styling of the mobile menu.
3. Implemented smooth scrolling to the corresponding page section when a mobile menu option is clicked.

### **Testing:**
Tested the implemented features on various emulated mobile devices to ensure proper functionality and responsiveness.

Please review this pull request. It was a collaborative effort, and I would like to acknowledge the contributions of my partner, Zuheb, in implementing these features.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const { target } = event;
+  const sectionId = target.getAttribute('href').substring(1);
+
+  toggleMobileMenu();
+
+  document.querySelector(sectionId).scrollIntoView({
+    behavior: 'smooth',
+    block: 'start',
+  });
+}
```
Comment (didierganthier) [inline script.js:24]: Well done on implementing the navigation making the menu navigate to the right section when an item is clicked 💯 but I noticed that the menu is still open after navigating, please ensure that the menu closes after that to provide a better user experience. 📱 

You can do that by adding an event listener to each one of these and making sure that the menu closes when they're clicked. Read more about event listeners [here](https://www.w3schools.com/js/js_htmldom_eventlistener.asp). 🔗 

In your case here you can just call the function to close the menu ❌

File: index.html [hunk 2]
```diff
@@ -9,6 +9,16 @@
...
     <div class="home-bg">
+      <div class="mobile-menu">
+        <div class="close-button">
+          <img src="./assets/close-button.png" alt="">
+        </div>
+        <ul class="mobile-menu-items">
+          <li class="mobile-menu-item"><a href="#portfolio" class="mobile-menu-link">Portfolio</a></li>
+          <li class="mobile-menu-item"><a href="#about" class="mobile-menu-link">About</a></li>
+          <li class="mobile-menu-item"><a href="#contact" class="mobile-menu-link">Contact</a></li>
+        </ul>
```
Comment (didierganthier) [inline index.html:20]: Good job so far 👍🏽 but the rest of the page is showing while the menu is open, you can either make it take the full height of the screen or just make the rest of the page unscrollable while it's opened 😉 take a look at the design issue it gives 👇🏽 

Screenshot 2023-05-16 at 07 31 34

**Review by didierganthier — CHANGES_REQUESTED** [review]
Hi @Peter-Kibet and @zuhebahmed88091 ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.



## Highlights
- Mobile menu showing ✔️ 
- Close button working ✔️ 
- Followed Figma template ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- N/A

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

**Review by AbelG101 — APPROVED** [review]
# Hi @Peter-Kibet 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Correct use of GitHub Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: styles.css [hunk 3]
```diff
@@ -75,6 +75,52 @@
...
+    left: 0;
+    top: 0;
+    background-color: #fff;
+    flex-direction: column;
+    justify-content: flex-start;
+    align-items: flex-start;
+    padding-left: 25px;
+    z-index: 101;
+    display: none;
+  }
```
[the remark this answers is not in the document]
Comment (Peter-Kibet) [reply styles.css:91]: Okay, Thanks for this, I will work on it as soon as possible.

**Pull request merged**